Data/Primitive/SmallArray.hs

```diff
@@ -797,3 +848,25 @@ instance (Typeable s, Typeable a) => Data (SmallMutableArray s a) where
   gunfold _ _ = die "gunfold" "SmallMutableArray"
   dataTypeOf _ = mkNoRepType "Data.Primitive.SmallArray.SmallMutableArray"
 #endif
+
+smallArrayFromListN :: Int -> [a] -> SmallArray a
```

Comment:
Since we're exporting this, it would be good to add at least some documentation for what this is doing.

Reply:
Will do.